stko/calculators/torsion_calculators.py

```diff
@@ -135,3 +137,74 @@ def get_results(self, mol):
             generator=self.calculate(mol),
             mol=mol,
         )
+
+
+class MatchedTorsionCalculator(ConstructedMoleculeTorsionCalculator):
+    """Matches rdkit generated torsions with building block torsions.
+    """
+
+    def calculate(self, mol: stk.ConstructedMolecule):
+        """extract torsions with rdkit, then match to building blocks
+
+        This method loops through each rdkit generated torsion. For
+        each torsion, it checks if the two interior atoms of the
+        torsion come from the two interior (central) atoms of some
+        torsion in a building block. If so, it replaces the end atoms
+        of the torsion with the atoms corresponding to the end atoms
+        of the underlying building block torsion.
+
+        Parameters
+        ----------
+        mol : :class:`.ConstructedMolecule`
+            The :class:`.ConstructedMolecule` whose torsions are to be
+            calculated.
+
+        """
+
+        torsions = list(next(super().calculate(mol)))
+        atom_maps = stko.get_atom_maps(mol)
+
+        # loop over torsions, updating each to match a building block
+        # torsion if possible
+        for i, torsion in enumerate(torsions):
+            # atom ids, atom infos, and building block ids of current
+            # torsion in constructed molecule
+            atom_ids = list(torsion.get_atom_ids())
+            atom_infos = list(mol.get_atom_infos(atom_ids))
+            build_block_ids = [atom_info.get_building_block_id()
+                               for atom_info in atom_infos]
+
+            # check if two central atoms of torsion are from the same
+            # building block. if not, leave this torsion alone
+            if build_block_ids[1] is None:
```

Comment:
The comment above does not correspond to this check - right?

Write the comments in order of the checks.